index/scheduler.py

```diff
@@ -0,0 +1,137 @@
+from collections import deque
+
+def noConflict(section,schedule):
+    '''
+    noConflict - Checks if the given section conflicts with any other sections in the schedule
+    Params: {Section} section - The section to check against
+    {[[string, Section]]} schedule - An array of string-Section pairs representing the current schedule
+    Returns: {boolean} - TRUE if the section does not conflict with any other section in the array, FALSE otherwise
+    '''
+    for scheduleSection in schedule:
+        if hasTimeConflict(section, scheduleSection):
+            return False
+    return True
+
+def hasTimeConflict(section1, section2):
+    '''
+    Returns TRUE if there is a time conflict.
+    '''
+    #compare both sections' meeting times
+    for meetingTime1 in section1['meetingTimes']:
+        for meetingTime2 in section2['meetingTimes']:
+            #determine common weeks between sections
+            commonWeeks = set(meetingTime1['weeks']).intersection(set(meetingTime2['weeks']))
+
+            #if they have weeks in common, check days
+            if commonWeeks:
+                days1 = [meetingTime1['monday'],
+                        meetingTime1['tuesday'],
+                        meetingTime1['wednesday'],
+                        meetingTime1['thursday'],
+                        meetingTime1['friday'],
+                        meetingTime1['saturday'],
+                        meetingTime1['sunday']]
+                days2 = [meetingTime2['monday'],
+                        meetingTime2['tuesday'],
+                        meetingTime2['wednesday'],
+                        meetingTime2['thursday'],
+                        meetingTime2['friday'],
+                        meetingTime2['saturday'],
+                        meetingTime2['sunday']]
+            else:
+                return False
+
+            #element wise list of boolean AND between days1 and days2    
+            aAndB = [x and y for x,y in zip(days1,days2)]
+
+            #if they have days in common, check times
+            if any(aAndB):
+                if meetingTime1['startTime'] <= meetingTime2['startTime'] and meetingTime1['endTime'] >= meetingTime2['startTime']:
+                    return True
+                if meetingTime2['startTime'] <= meetingTime1['startTime'] and meetingTime2['endTime'] >= meetingTime1['startTime']:
+                    return True
```

Comment:
Logically these work out. I'm curious whether there is a way to do it with fewer operations, considering the first `<=` comparisons on each line are just flipped.